localstack/services/apigateway/apigateway_listener.py

```diff
@@ -118,9 +118,12 @@ def forward_request(self, method, path, data, headers):
                     response = Response()
                     parsed_result = json.loads(result)
                     response.status_code = int(parsed_result['statusCode'])
-                    response.headers.update(parsed_result['headers'])
-                    response_body = parsed_result['body']
-                    response._content = json.dumps(response_body)
+                    response.headers.update(parsed_result.get('headers', {}))
+                    try:
+                        response_body = parsed_result['body']
```

Comment:
I've actually just tested this, and `parsed_result.get('body', '')` still provides an empty object return, so the way it's written now is the best implementation.